kivy/core/text/__init__.py

```diff
@@ -263,11 +271,7 @@ def __init__(
         kwargs_get = kwargs.get
         options['color'] = color or (1, 1, 1, 1)
         options['outline_color'] = outline_color or (0, 0, 0, 1)
-        options['padding'] = kwargs_get('padding', (0, 0))
-        if not isinstance(options['padding'], (list, tuple)):
-            options['padding'] = (options['padding'], options['padding'])
-        options['padding_x'] = kwargs_get('padding_x', options['padding'][0])
-        options['padding_y'] = kwargs_get('padding_y', options['padding'][1])
+        options['padding'] = kwargs_get('padding', (0, 0, 0, 0))
```

Comment:
`LabelBase` should be considered as part of the public API, and as `Label`, support for `padding_x` and `padding_y` needs to be deprecated, but not removed.

A migration path should also be applied to `padding`. If a user specifies an integer, instead of the new format, we need to print a deprecation warning and then convert the integer to the new format.